apps/dcos_l4lb/src/dcos_l4lb_ipset_mgr.erl

```diff
@@ -0,0 +1,362 @@
+-module(dcos_l4lb_ipset_mgr).
+
+-include_lib("gen_netlink/include/netlink.hrl").
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+    get_entries/0,
+    get_entries/1,
+    add_entries/2,
+    remove_entries/2,
+    cleanup/0
+]).
+
+%% gen_server callbacks
+-export([init/1, handle_info/2, terminate/2,
+    code_change/3, handle_call/3, handle_cast/2]).
+
+-export_type([entry/0]).
+
+-type entry() :: {protocol(), inet:ip_address(), inet:port_number()}.
+
+-record(state, {
+    netlink :: pid()
+}).
+
+-define(IPSET_PROTOCOL, 6).
+-define(IPSET_TYPE, "hash:ip,port").
+-define(IPSET_NAME, "dcos-l4lb").
+-define(IPSET_NAME_IPV6, "dcos-l4lb-ipv6").
+
+
+-spec(get_entries() -> [entry()]).
+get_entries() ->
+    % NOTE: FOR DEBUG PURPOSE ONLY.
+    {ok, Pid} = start_link(),
+    try
+        get_entries(Pid)
+    after
+        erlang:unlink(Pid),
+        gen_server:cast(Pid, stop)
+    end.
+
+-spec(get_entries(pid()) -> [entry()]).
+get_entries(Pid) ->
+    gen_server:call(Pid, get_entries).
+
+-spec(add_entries(pid(), [entry()]) -> ok).
+add_entries(Pid, Entries) ->
+    gen_server:call(Pid, {add_entries, Entries}).
+
+-spec(remove_entries(pid(), [entry()]) -> ok).
+remove_entries(Pid, Entries) ->
+    gen_server:call(Pid, {remove_entries, Entries}).
+
+-spec(start_link() ->
+    {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link() ->
+    gen_server:start_link(?MODULE, [], []).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+init([]) ->
+    self() ! init,
+    {ok, dcos_l4lb_config:ipset_enabled()}.
+
+handle_call(get_entries, _From, disabled) ->
+    {reply, [], disabled};
+handle_call(_Request, _From, disabled) ->
+    {reply, ok, disabled};
+handle_call(get_entries, _From, #state{netlink=Pid}=State) ->
+    {ok, EntriesIPv4} = get_entries(Pid, ?IPSET_NAME),
+    {ok, EntriesIPv6} = get_entries(Pid, ?IPSET_NAME_IPV6),
+    {reply, EntriesIPv4 ++ EntriesIPv6, State};
+handle_call({add_entries, Entries}, _From, #state{netlink=Pid}=State) ->
+    lists:foreach(fun ({Protocol, IP, Port}) ->
+        {ok, []} = add_entry(Pid, Protocol, IP, Port)
+    end, Entries),
+    {reply, ok, State};
+handle_call({remove_entries, Entries}, _From, #state{netlink=Pid}=State) ->
+    lists:foreach(fun ({Protocol, IP, Port}) ->
+        {ok, []} = del_entry(Pid, Protocol, IP, Port)
+    end, Entries),
+    {reply, ok, State};
+handle_call(_Request, _From, State) ->
+    {noreply, State}.
+
+handle_cast(stop, State) ->
+    {stop, normal, State};
+handle_cast(_Request, State) ->
+    {noreply, State}.
+
+handle_info(init, false) ->
```

Comment:
Note: I had to move it to `handle_info/2` because OTP 19 doesn't support for `handle_continue/2`.